File: Asphalt/Asphalt.js
function potholePatchCounter(S) {
    const N = S.length;
    let patches = 0;

    for (let i = 0; i < N; i++) {
        if (S[i] === 'X') {
            patches++;
            i += 2; 
        }
    }

    return patches;
}


console.log(potholePatchCounter(".X..X"));       
console.log(potholePatchCounter("X.XXXXX.X."));   
console.log(potholePatchCounter("XX.XXX.."));      
console.log(potholePatchCounter("XXXX"));          
console.log(potholePatchCounter('.X...XX'));       

File: Asphalt/Asphalt.py
def solution(S):
    count = 0
    B = len(S)
    i = 0

    while i < B: 

         if S[i] == "X":
            count +=1
            i += 3
         else:
             i+=1

    return count
     



print(solution(".X..X"))         
print(solution("X.XXXXX.X."))    
print(solution("XX.XXX.."))      
print(solution("XXXX"))          
print(solution('.X...XX'))       

